test/integer.cpp

```diff
@@ -295,4 +295,108 @@ TEST_CASE("integer")
         ia = type_safe::abs(i);
         REQUIRE(static_cast<unsigned>(ia) == 123u);
     }
+    SECTION("signed_to_unsigned")
+    {
+        // From int8_t
+        REQUIRE( !std::is_convertible<integer<int8_t>, integer<uint8_t>>::value );
+        REQUIRE(  std::is_convertible<integer<int8_t>, integer<uint16_t>>::value );
```

Comment:
This test says that we can convert `int8_t` to `uint16_t`, which is wrong because of negative numbers.

Reply:
This is also the cause of the warning.